### ISSUE TITLE
Self-edges!

### ISSUE THREAD

**Issue by nschneid** [amrisi/amr-guidelines#234]
I didn't think these were allowed, but here is an example where "X when and if X takes place" is annotated with `X :time X :condition X`

A lot of no-votes guys and here they are voting on something like this and here they're going to be voting on it again when and if that vote ever takes place.

```
(s2 / say-01 
      :ARG1 (a / and 
            :op1 (p / person 
                  :ARG0-of (v / vote-01 
                        :ARG1 (h / have-polarity-91 
                              :ARG2 -) 
                        :quant (l / lot))) 
            :op2 (v2 / vote-01 
                  :ARG0 p 
                  :ARG1 (s / something 
                        :ARG1-of (r / resemble-01 
                              :ARG2 (t / this))) 
                  :location (h2 / here)) 
            :op3 (v3 / vote-01 
                  :ARG0 p 
                  :ARG1 s 
                  :location h2 
                  :mod (a2 / again) 
                  :time v3 
                  :condition v3 
                  :time (e / ever))) 
      :ARG2 (g / guy))
```

**Comment by goodmami** [comment]
But these ["magically disappear upon reification"](https://github.com/amrisi/amr-guidelines/blob/master/amr.md#cycles), just as the cycle does, yes?

```
(s2 / say-01 
      :ARG1 (a / and 
            :op1 (p / person 
                  :ARG0-of (v / vote-01 
                        :ARG1 (h / have-polarity-91 
                              :ARG2 -) 
                        :quant (l / lot))) 
            :op2 (v2 / vote-01 
                  :ARG0 p 
                  :ARG1 (s / something 
                        :ARG1-of (r / resemble-01 
                              :ARG2 (t / this))) 
                  :location (h2 / here)) 
            :op3 (v3 / vote-01 
                  :ARG0 p 
                  :ARG1 s 
                  :location h2 
                  :mod (a2 / again) 
                  :ARG1-of (t2 / be-temporally-at-91 :ARG2 v3) 
                  :ARG1-of (c / have-condition-91 :ARG2 v3) 
                  :time (e / ever))) 
      :ARG2 (g / guy))
```

Although in the above, it doesn't seem to matter if you attach the reified nodes via `:ARG1-of` or `:ARG2-of`.

**Comment by nschneid** [comment]
Sure, they disappear on reification because reification turns the edges into nodes.

But I don't think I agree with the linguistic analysis in the first place. It seems there are two aspects of voting being referred to: (i) having the opportunity to cast a vote, and (ii) casting a vote for or against a proposal. The only sense I can make of "they're going to be voting on it again when and if that vote ever takes place" is that "they're going to be voting on it again" refers to (ii) while "when and if that vote ever takes place" refers to (i)—roughly, they're going to vote (cast ballots) when there is a vote (a call to the legislative body to cast ballots). (ii) has an ARG0, (i) doesn't. So I think I'd use two different `vote-01` frames instead of the self-edge.

**Comment by goodmami** [comment]
Sorry, I didn't make my point very clear. From the guidelines regarding whether to use reification or not:

> The resolution: we consider “AMR with reification” to be “real AMR”, with non-reified relations as semantic sugar. Therefore, if you are translating English into AMR, the rule is “whenever you feel like it”, because your AMRs will be normalized into reified form behind the scenes.

According to this statement, the self-edges you're seeing are just the "semantic sugar" and not the real graph. I'm curious how accurate the statement is because, for instance, smatch does **not** give 1.0 for two AMRs that differ only in reification.

Regarding the analysis... there's a lot not to love about that AMR (where does `(s2 / say-01 ... :ARG2 (g / guy))` come from? Maybe I'm missing context.), and the original sentence isn't terribly clear. I think you're right about the (i) vs (ii) interpretations of "vote" here.

**Comment by nschneid** [comment]
> I'm curious how accurate the statement is because, for instance, smatch does not give 1.0 for two AMRs that differ only in reification.

Yeah, my impression was always that the policy was "as annotators let's feel free to reify without guilt", but smatch does not actually handle the equivalence in practice.

> where does (s2 / say-01 ... :ARG2 (g / guy)) come from?

I think "A lot of no votes guys" has been interpreted as vocative. say-01 is sometimes introduced for speech acts such as vocatives and quotations. (In retrospect, we probably should have used a different frame for those.)

**Comment by nschneid** [comment]
Oh yeah, that is strange. Right now it looks like "there is/are a person/people who vote no a lot", which can't be right. Either `vote-01` should be focused ("there is a lot of no-voting") or the `:quant` should be under `person` ("there are a lot of no-voters").
